plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/AbstractPinotIntegrationSmokeTest.java

```diff
@@ -398,6 +510,57 @@ protected QueryRunner createQueryRunner()
         return ImmutableMap.of();
     }
 
+    private static Path createSegment(InputStream tableConfigInputStream, InputStream pinotSchemaInputStream, RecordReader recordReader, String outputDirectory, int sequenceId)
```

Comment:
Would now be a good time to have a PinotTestingUtils to hold all the data-setup? (No change requested yet)

(Note to self: need to still look at this method)

Reply:
Nice! Should it be a separate pr? Might also make sense to put it into insert pr (updating now) since insert will use similar code? lmk what you think.

Reply:
It can be a separate PR (or part of some future PR that isn't yet reviewed - INSERT?)